### PR TITLE
Fix some typos/language in `regex` and `sequence`

### PR DESCRIPTION
**Pull request by jrysana**
Please correct me if I'm wrong about any of the below! But I noticed a few more things that seem mismatched in comments.

- [x] Fixed a typo in `regex/create_proposal` implying that this method is only for integer generation, whereas it seems clear that this method is used for all schemas.
- [x] Fixed some typos where argument names were mismatched or nonexistent in the actual function code.
- [x] Fixed some typos for overall language (e.g. "used to compute" instead of "used to computes") and to clarify potentially confusing comments.
- [x] Fixed a broken link in `README.md`

### CODE REVIEW

File: outlines/text/generate/regex.py
Comment (jrysana) [inline outlines/text/generate/regex.py:214]: This is a nitpick, but I think these comments for `model` args might be more welcoming to some readers with less context, without losing clarity, if we said "the language model" / "the text transformer model" or something, instead of just "the model" - being that "model" is a very vague word and requires previous context to understand. For example, just in this file alone, there is a connotation of "model" as a text transformer model and "model" as a Pydantic data validation model.

Comment (zaqqwerty) [reply outlines/text/generate/regex.py:214]: Hey @jrysana, that sounds like a good idea to me 🙂  would you want to make those changes?

**Review by zaqqwerty — APPROVED** [review]
Hi @jrysana, thank you for the PR!  We appreciate your thorough reading of the code.  LGTM, lmk if you'd first like to update based on your suggestion before we merge.

File: outlines/text/generate/regex.py
Comment (jrysana) [reply outlines/text/generate/regex.py]: Sure, pushed @zaqqwerty

**Pull request merged**
